### PR TITLE
docs: Fix description of tls options

### LINKED ISSUE #10388: docs: confusing tls description for sinks
The various sinks that mention TLS, for example [the http sink](https://vector.dev/docs/reference/configuration/sinks/http/) or the [loki sink](https://vector.dev/docs/reference/configuration/sinks/loki/) say

> Configures the TLS options for incoming connections.

... when the sink doesn't have options for incoming connections.

It made it unclear if the sink support client certificates for TLS.

### PR DESCRIPTION
**Pull request by spencergilbert**
Signed-off-by: Spencer Gilbert <spencer.gilbert@datadoghq.com>

Closes #10388

### CODE REVIEW

File: website/cue/reference/components.cue [hunk 1]
```diff
@@ -625,7 +625,7 @@ components: {
 				let Args = _args
 
 				common:      false
-				description: "Configures the TLS options for incoming connections."
+				description: "Configures the TLS options for outgoing connections."
```
Comment (binarylogic) [inline website/cue/reference/components.cue:628]: Won't this be wrong for sources?

Comment (spencergilbert) [reply website/cue/reference/components.cue:628]: It _seemed_ like sources would be using: https://github.com/vectordotdev/vector/blob/756f16ec51b892ce30159b7b7ac2309f760bdfb2/website/cue/reference/components.cue#L549

Comment (spencergilbert) [reply website/cue/reference/components.cue:628]: https://deploy-preview-10394--vector-project.netlify.app/docs/reference/configuration/sources/http/#tls

vs 

https://deploy-preview-10394--vector-project.netlify.app/docs/reference/configuration/sinks/http/#tls

**Review by jszwedko — APPROVED** [review]
Thanks!

**Pull request merged**